modules/ShipStation/Api/src/Controller/Customer/Api.php

```diff
@@ -80,66 +80,70 @@ public function getAction()
-                            if ($objShippingMethod->getName()) {
-                                $objOrder->setShippingMethodName($objShippingMethod->getName());
-                            }
-                        }
-                    }
+        if (!isset($_GET['order_number']) || !$_GET['order_number']) {
+            echo 'No order number found in action';
+            return;
+        }
+
+        /**
+         * Check if the order is present in xcart
+         */
+        $objOrder = \XLite\Core\Database::getRepo('XLite\Model\Order')->findOneByOrderNumber(intval($_GET['order_number']));
+        if (!$objOrder) {
+            echo 'Order does not exist in database';
+            return;
+        }
+
+        /**
+         * Change the order shipping status to shipped
+         * Change the order payment status to paid
+         * Change the Shipping Method based on service code recived
+         */
+        $objOrder->setShippingStatus(\XLite\Model\Order\Status\Shipping::STATUS_SHIPPED);
+        $objOrder->setPaymentStatus(\XLite\Model\Order\Status\Payment::STATUS_PAID);
+        if (!isset($_GET['shipperServiceID']) || !$_GET['shipperServiceID']) {
+            return;
+        }
+
+        // Check the shipstation service mapping with the X -Cart.
+        $serviceCode = \XLite\Core\Database::getRepo('XLite\Model\Order')->getShippingServices($_GET['shipperServiceID']);
+        
+        if ($serviceCode) {
+            $objShippingMethod = \XLite\Core\Database::getRepo('XLite\Model\Shipping\Method')->findOneBy(
+                array('code' => $serviceCode)
+            );
+            if (!empty($objShippingMethod)) {
+                if ($objShippingMethod->getMethodId()) {
+                    $objOrder->setShippingId($objShippingMethod->getMethodId());
                 }
-                /**
-                 * Add tracking details
-                 */
-                $arrTrackingValues = array();
-                if ($objOrder->getOrderId() && $_GET['comment']) {
-                    foreach ($objOrder->getTrackingNumbers()as $objTrackingNumber) {
-                        $arrTrackingValues[] = $objTrackingNumber->getValue(); //get the tracking number of order
-                    }
-                    if (!in_array($_GET['comment'], $arrTrackingValues))
-                        \XLite\Core\Database::getRepo('XLite\Model\Order')->addOrderTrackingNumber($objOrder->getOrderId(), $_GET['comment']);
+                if ($objShippingMethod->getName()) {
+                    $objOrder->setShippingMethodName($objShippingMethod->getName());
                 }
+            }
+        }
 
-                \XLite\Core\Database::getEM()->flush(); //clear the cache
-
-                echo 'Status updated successfully';
-            } else {
-                echo 'Order does not exist in database';
+        /**
+         * Add tracking details
+         */
+        $arrTrackingValues = array();
+        $trackingID = $_GET['comment'];
```

Comment:
Apparently we also use `$_GET['comment']` to pass the TrackingID in the URL to the function. I added a variable to help clarify a little.